include/circt/Dialect/ESI/ESITypes.td

```diff
@@ -29,6 +29,21 @@ def ESIAnyType : ESI_Type<"Any"> {
   }];
 }
 
+def ESIListType : ESI_Type<"List"> {
+  let summary = "a runtime-variably sized list";
+  let description = [{
+    In software, a chunk of memory with runtime-specified length. In hardware, a
+    stream of runtime-specified amount of data transmitted over many cycles in
+    compile-time specified specified windows (chunks).
```

Comment:
Where is that runtime-specified length supposed to be specified? is it a value that can be extracted from an SSA value of `!esi.list` type?

> compile-time specified specified windows
Similar to the above - where is this defined?

How is this intended to compose with `!esi.channel`?

Reply:
> is it a value that can be extracted from an SSA value of `!esi.list` type?

Yes

> > compile-time specified specified windows
> 
> Similar to the above - where is this defined?

https://github.com/llvm/circt/blob/cc627a0889c3df3e0839c235acb9bb626afc2c3c/include/circt/Dialect/ESI/ESITypes.td#L33



> How is this intended to compose with `!esi.channel`?

`!esi.channel<!esi.list<int>>`

Reply:
This first take is pretty exclusively for modeling only. So we can add it to our generated APIs and (in the future) it can be used by CAM. Totally unusable in actual hardware without data windows and data window lowering to hw.